dm/master/openapi.go

```diff
@@ -88,17 +90,64 @@ func (s *Server) GetDocHTML(ctx echo.Context) error {
 
 // DMAPICreateSource url is:(POST /api/v1/sources).
 func (s *Server) DMAPICreateSource(ctx echo.Context) error {
-	return nil
+	needRedirect, host, err := s.redirectRequestToLeader(ctx.Request().Context())
+	if err != nil {
+		return err
+	}
+	if needRedirect {
+		return ctx.Redirect(http.StatusTemporaryRedirect, fmt.Sprintf("http://%s%s", host, ctx.Request().RequestURI))
+	}
+
+	var createSourceReq openapi.Source
+	if err := ctx.Bind(&createSourceReq); err != nil {
+		return err
+	}
+	cfg := modelToSourceCfg(createSourceReq)
+	if err := checkAndAdjustSourceConfig(ctx.Request().Context(), cfg); err != nil {
+		return err
+	}
+	if err := s.scheduler.AddSourceCfg(cfg); err != nil {
+		return err
+	}
```

Comment:
`OperateSource` has waited for that worker has responsed to the action

Reply:
At the API level, I don't think it's necessary to return whether the source is successfully bound or not, the user can wheel through the `source status`(in next PR) api to get the result

Essentially because the DM implementation is declarative and asynchronous, the success of the API is simply an indication that dm-master has accepted the request and started processing it

Reply:
OK, if TiCDC also treat the API as not declarative, we can also do so.

Reply:
after read cdc's code,  it did the same https://github.com/pingcap/ticdc/blob/8abf12a5f29143d0379103847eb2b08d9a033524/cdc/capture/http_handler.go#L234